<launch>
  <arg name="image"/>
  <arg name="autostart" default="True" />
  <arg name="num_rows" default="6" />
  <arg name="num_cols" default="8" />
  <arg name="dim" default="1" />
  <arg name="flags" default="0" />

  <arg name="viewer" default="True" />
  <arg name="marker_size" default="-1"/>
  <arg name="marker_width" default="-1"/>
  <arg name="scaling" default="-1" />
  
  <node type="calibration_detector"
        pkg="image_cb_detector"
        name="checkerboard_detector"
        args="_autostart:=$(arg autostart)"
        output="screen"
        required="true">
    <remap from="image_raw" to="$(arg image)" />
  </node>

  <node type="image_annotator"
        pkg="image_cb_detector"
        name="calibration_annotator"
        output="screen">
    <remap from="image" to="calibration_image" />
    <param name="marker_size" type="int" value="$(arg marker_size)" />
    <param name="marker_width" type="int" value="$(arg marker_width)" />
    <param name="scaling" type="double" value="$(arg scaling)" />
  </node>

  <node if="$(arg viewer)" type="image_view" pkg="image_view" name="$(anon image_view)">
    <remap from="image" to="annotated" />
  </node>
</launch>
